seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/sink/StarRocksSinkFactory.java

```diff
@@ -89,13 +89,15 @@ public TableSink createSink(TableSinkFactoryContext context) {
         String sinkDatabaseName = sinkConfig.getDatabase();
         String sinkTableName = sinkConfig.getTable();
         // to replace
-        String finalDatabaseName =
-                sinkDatabaseName.replace(REPLACE_DATABASE_NAME_KEY, sourceDatabaseName);
+        if (StringUtils.isNotBlank(sourceDatabaseName)) {
+            sinkDatabaseName =
+                    sinkDatabaseName.replace(REPLACE_DATABASE_NAME_KEY, sourceDatabaseName);
+        }
```

Comment:
I prefer when source database name is null, we can treat it as empty string.

```suggestion
sinkDatabaseName =                  sinkDatabaseName.replace(REPLACE_DATABASE_NAME_KEY, sourceDatabaseName != null ? sourceDatabaseName: "");
```

Reply:
Done